File: src/main/java/net/dontdrinkandroot/fixtures/loader/FixtureLoader.java
package net.dontdrinkandroot.fixtures.loader;

import net.dontdrinkandroot.fixtures.Fixture;
import net.dontdrinkandroot.fixtures.ReferenceRepository;

import java.util.Collection;

/**
 * A {@link FixtureLoader} loads a given set of fixtures into the database.
 *
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public interface FixtureLoader
{
    /**
     * Loads the given fixtures.
     *
     * @param fixtureClasses The fixtures to load.
     * @return The populated {@link ReferenceRepository}
     */
    ReferenceRepository load(Collection<Class<? extends Fixture>> fixtureClasses);
}

File: src/main/kotlin/net/dontdrinkandroot/fixtures/loader/FixtureLoader.kt
package net.dontdrinkandroot.fixtures.loader

import net.dontdrinkandroot.fixtures.Fixture
import net.dontdrinkandroot.fixtures.referencerepository.ReferenceRepository
import kotlin.reflect.KClass

/**
 * A [FixtureLoader] loads a given set of fixtures into the database.
 */
interface FixtureLoader {

    /**
     * Loads the given fixtures.
     *
     * @param fixtureClasses The fixtures to load.
     * @return The populated [ReferenceRepository]
     */
    fun load(fixtureClasses: Collection<KClass<out Fixture>>): ReferenceRepository
}